### ISSUE TITLE
IOB is calculated wrong when temp basal is crossing base basal rate change

### ISSUE THREAD

**Issue by MilosKozak** [openaps/oref0#448]
![image](https://cloud.githubusercontent.com/assets/11146458/25498928/05c87d02-2b8a-11e7-8532-436758922fe9.png)
changed algorithm following way:
split temp basals to about 5 min intervals and calculate contribution to middle of the interval with basal rate in this time
https://github.com/MilosKozak/AndroidAPS/commit/0aad00808c02a400b4af2acf68c612b94965766c
![image](https://cloud.githubusercontent.com/assets/11146458/25498955/21d842ac-2b8a-11e7-8160-5db929792776.png)

**Comment by thebookins** [comment]
Is this fixed with use of mmistorytools (https://github.com/openaps/oref0/pull/375)?

**Comment by sulkaharo** [comment]
I fixed this last summer and even put unit tests in for this - @MilosKozak can you add a test into the OpenAPS repo that fails due to the issue?

**Comment by sulkaharo** [comment]
Checked, the fix I put in is in place; the temp basal is split both on the ~5 minute intervals but also on the moments where the base basal rate changes, so the processed temp never crosses the base change when calculating the IOB contribution of the temp. If this is producing wrong results, we need to look into the part of the code that calculates the IOB contribution of the generated 5-minute microboluses. I recall the IOB calculation did break if I adjusted the length of the 5-minute split (while it should not have).
